### PR TITLE
fix: added semantic release workflow

### PR DESCRIPTION
**Pull request by sahinvardar**
This adds semantic release workflow to the project. The release workflow will be trigged on changes on ```release``` branch. In addition I also added PR build and test workflow, so that each PR has the past build and test. 

Only projects in the ```packages``` directory will be published.

To trigger your first release you have to create a release branch. 

I deliberately chose the ```fix``` prefix in the commit message to trigger version bump to trigger the release

### CODE REVIEW

File: release.config.cjs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/**
+ * @type {import('semantic-release').GlobalConfig}
+ */
+module.exports = {
+  branches: ['release']
```
Comment (joemays) [inline release.config.cjs:5]: Seems simpler to put this in the package.json file perhaps?
```json
{
  "release": {
    "branches": ["release"]
  }
}
```

**Review by joemays — COMMENTED** [review]
Looks great, a few specific questions in the review, and a few general questions below:

1. Should we be using something like [commitlint](https://github.com/conventional-changelog/commitlint) or [commitizen](https://github.com/commitizen/cz-cli) to make sure the commits follow proper convention?
2. How can I start with 0.1.0?  Something like [this](https://semantic-release.gitbook.io/semantic-release/usage/configuration#existing-version-tags)?
3. Aside from creating release branch, what are the next steps to getting this set up and publish 0.1.0?

File: packages/client/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@apimda/client",
-  "version": "0.0.11",
+  "version": "0.0.0-development",
```
[the remark this answers is not in the document]
Comment (joemays) [reply packages/client/package.json:3]: Never mind, just read that semantic-release does not update package.json by default.  I'm very okay with that.

File: package.json [hunk 3]
```diff
@@ -26,6 +27,8 @@
     "prettier": "^2.8.8",
     "prettier-package-json": "^2.8.0",
     "prettier-plugin-organize-imports": "^3.2.2",
+    "semantic-release": "^22.0.10",
+    "semantic-release-monorepo": "^7.0.5",
```
[the remark this answers is not in the document]
Comment (sahinvardar) [reply package.json:31]: The monorepo dependency takes care to assign commits to the various packages, therefor it creates separate tags and releases for each package. As far as I know, I also don't see how we could handle the mono repo with semantic-release without the [semantic-release-monorepo](https://github.com/pmowrer/semantic-release-monorepo)

[the remark this answers is not in the document]
Comment (joemays) [reply package.json:31]: Okay, cool, thanks.  Kinda surprised that the original project didn't consider monorepos, but then again the entire JS ecosystem is still really figuring this out.

File: release.config.cjs (hunk 1 above)
Comment (sahinvardar) [reply release.config.cjs]: Good point, done

File: package.json [hunk 4]
```diff
@@ -11,6 +11,7 @@
     "lint": "prettier --plugin-search-dir . --check . && eslint \".\"",
     "playground": "pnpm --filter integration-test run playground",
     "prepare": "husky install",
+    "release": "pnpm -r --workspace-concurrency=1 exec -- npx --no-install semantic-release -e semantic-release-monorepo --filter ./packages",
```
[the remark this answers is not in the document]
Comment (sahinvardar) [reply package.json]: I removed the filter and npx command, in theory this should work.  I added the dry run for testing in CI so we can see what will happen

**Comment by sahinvardar** [comment]
> Looks great, a few specific questions in the review, and a few general questions below:
> 
> 1. Should we be using something like [commitlint](https://github.com/conventional-changelog/commitlint) or [commitizen](https://github.com/commitizen/cz-cli) to make sure the commits follow proper convention?
> 2. Aside from creating release branch, what are the next steps to getting this set up and publishing the first release?
> 
> Edit: I re-read the semver spec, and think it's best to start at 1.0.0.

1. Good point, would help to make sure we have proper commit messages. I can add this to this PR or make a new one after that one. Open for both options, what aver suites the best.

2. To start with 1.0.0 we have to add ```BREAKING CHANGE:``` in the footer of the resulting commit of this PR.
That it, as soon as we have a release branch the work flow for release will trigger automatically

**Review by joemays — APPROVED** [review]
Okay, sorry this took so long.  I think we're good.

**Pull request merged**